cli/defenseclaw/commands/cmd_doctor.py

```
@@ -4751,92 +6136,477 @@ def _fix_gateway_token_drift(cfg, *, assume_yes: bool) -> tuple[str, str]:
+    if acl_read_problem:
+        prompt = f"    Remove the read-capable extended ACL from {path} and enforce 0600?"
+    if not assume_yes and not click.confirm(prompt, default=True):
         return ("skip", "declined by user")
 
     try:
-        os.chmod(path, 0o600)
+        protect_private_file(path)
+        remaining_acl_problem = darwin_acl_confidentiality_error(path) if platform_name == "darwin" else None
+        if remaining_acl_problem is not None:
+            return ("fail", f"dotenv extended ACL remains unsafe: {remaining_acl_problem}")
         return ("pass", f"set {path} to 0600")
```

Ruff (0.16.0)

[warning] Consider moving this statement to an `else` block

(TRY300)